openrtx/src/rtx/rtx.cpp

```diff
@@ -117,15 +127,63 @@ void rtx_task()
         {
             // Copy new configuration and override opStatus flags
             uint8_t tmp = rtxStatus.opStatus;
+            // #ifdef CONFIG_SPECTRUM
+            // rxSweep_t tmp_sweep;
+            // tmp_sweep = rtxStatus.rxSweep_data; // Save current sweep data
+            // #endif
             memcpy(&rtxStatus, newCnf, sizeof(rtxStatus_t));
             rtxStatus.opStatus = tmp;
+            // #ifdef CONFIG_SPECTRUM
+            // rtxStatus.rxSweep_data = tmp_sweep; // Restore current sweep data
+            // #endif
 
             reconfigure = true;
             newCnf = NULL;
         }
 
         pthread_mutex_unlock(cfgMutex);
     }
+     /* Spectrum update block, run when in SPECTRUM mode.
+    *
+    * The spectrum mode is a special mode where the radio is in RX mode but
+    * the audio path is disabled. This allows to display the RSSI level of the
+    * received signals across a frequency range in a waterfall-like display.
+    * 
+    * This block writes the received RSSI levels to the spectrum buffer.
+    */
+    #ifdef CONFIG_SPECTRUM
+    if(state.rtxStatus == RTX_RX_SWEEP)
+    {
+        rtxStatus.rxSweep_data.sweepDone = false; // Clear the flag at the start of the sweep
+        uint32_t spectrumStep = freq_steps[state.settings.spectrum_step];
+        // Get the current RSSI level
+        // uint8_t peakIndex; // No longer needed
+        rtxStatus.rxSweep_data.peakRssi = -160; // Reset peak RSSI for each sweep
+        // Write the RSSI level to the spectrum buffer
+        uint32_t spanWidth = RX_SWEEP_POINTS * freq_steps[state.settings.spectrum_step];
+        uint32_t startFreq = state.rxSweep_start_freq;
+        for (int i = 0; i < RX_SWEEP_POINTS; i++) {
```

Comment:
Does it make sense to have this scan part doing one step for each update of the rtx_task? In FM mode the update rate is 33Hz, so it will take around 2s to do a full scan

Reply:
I'm not sure I understand. As far as I can tell, it does a full sweep for every one execution of the rtx_task. Or did you mean that it _should_ do one step for each rtx_task update? I think 2s is too long for one full scan; currently, it updates 2-4 times a second.